src/sentry/lang/javascript/processor.py

```diff
@@ -712,8 +732,117 @@ def bind_release(self, release=None, dist=None):
+            # multi-join query. The double query solution is better if we are confident of a high-level of cache hits
+            # but considering that this code will be moved to symbolicator where a completely new caching system is in
+            # place, we shouldn't worry too much.
+            debug_id_artifact_bundle = self._get_debug_id_artifact_bundle_entry(
+                debug_id, source_file_type
+            )
+            artifact_bundle = debug_id_artifact_bundle.artifact_bundle
+            bundle_id = artifact_bundle.bundle_id
+
+            # We try to load the bundle file from the cache. The file we are loading here is the entire bundle, not
+            # the contents.
+            cache_key = get_artifact_bundle_cache_key(bundle_id)
+            result = cache.get(cache_key)
+            if result:
+                return BytesIO(result)
+
+            # We didn't find the bundle in the cache, thus we want to fetch it.
+            artifact_bundle_file = fetch_retry_policy(artifact_bundle.file.getfile)
+
+            # `cache.set` will only keep values up to a certain size,
+            # so we should not read the entire file if it's too large for caching
+            if (
+                CACHE_MAX_VALUE_SIZE is not None
+                and artifact_bundle_file.size > CACHE_MAX_VALUE_SIZE
+            ):
+                return artifact_bundle_file
+
+            with sentry_sdk.start_span(op="fetch_artifact_bundle.read_for_caching") as span:
+                span.set_data("file_size", artifact_bundle_file.size)
+                contents = artifact_bundle_file.read()
+            with sentry_sdk.start_span(op="fetch_artifact_bundle.write_to_cache") as span:
+                span.set_data("file_size", len(contents))
+                cache.set(cache_key, contents, 3600)
+
+            artifact_bundle_file.seek(0)
+            return artifact_bundle_file
+        except Exception as exc:
+            logger.error(
+                "Failed to load the artifact bundle for debug_id %s",
+                debug_id,
+                exc_info=exc,
+            )
+
+        return None
+
     def fetch_by_debug_id(self, debug_id, source_file_type):
-        # TODO: implement debug id lookup.
+        result_url = debug_id
+
+        # We try to load from the cache the data of the single file which was previously fetched with the combination
+        # ('debug_id', 'source_file_type'). This caching happens at the file level, thus we will load multiple
+        # times the same bundle if we want to fetch different files from it.
+        cache_key = get_artifact_bundle_file_cache_key(debug_id, source_file_type)
+        cache_key_meta = get_artifact_bundle_file_cache_key_meta(debug_id, source_file_type)
+        result = cache.get(get_artifact_bundle_file_cache_key(debug_id, source_file_type))
+        if result == -1:
```

Comment:
We should use the raw cache interface, but in that case the cache can only store bytes, so `-1` would no longer work.